### PR TITLE
React/week4

### PR DESCRIPTION
**Pull request by Saidemm**

### CODE REVIEW

File: react/week4/github-user-searcher/src/components/UserSearcherProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                    const errorResult = await apiData.json();
+                    console.log('Error in fetching data: ', errorResult.message);
+                    setError(errorResult);
+                }
+                setLoading(false);
+            } catch(error){
+                setError(error);
+            }
+        }
+        if(inputUser !== null && inputUser !== "" && inputUser !== lastInputUser.current){
```
Comment (calm-prog) [inline react/week4/github-user-searcher/src/components/UserSearcherProvider.jsx:38]: Nice input validation!

File: react/week4/github-user-searcher/src/components/UsersSearcher.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React, { useContext } from 'react';
+import DisplayUser from './DisplayUser.jsx';
+import {UserContext} from './UserSearcherProvider.jsx';
+
+function UsersSearcher(){
+    const ctxObj = useContext(UserContext);
```
Comment (calm-prog) [inline react/week4/github-user-searcher/src/components/UsersSearcher.jsx]: This ctxObj is a bit scripted variable name. I would name it something more descriptive.

File: react/week4/github-user-searcher/src/components/UsersSearcher.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                </div>
+            }
+            {ctxObj.loading && ctxObj.error === null && 
+                <div>
+                    Loading ....
+                </div>
+            }
+            {ctxObj.error !== null && 
+                <div>
+                    {ctxObj.error.message}
```
Comment (calm-prog) [inline react/week4/github-user-searcher/src/components/UsersSearcher.jsx]: Your error message appears nicely to the user, I've tested it as well. :+1:

**Review by calm-prog — COMMENTED** [review]
Really nice work! The app does exactly what it needs to, you really paid attention to implementing the correct behavior. I like how you structured your code, there is a nice separation in your logic that gives a good overview.

**Comment by Saidemm** [comment]
> Really nice work! The app does exactly what it needs to, you really paid attention to implementing the correct behavior. I like how you structured your code, there is a nice separation in your logic that gives a good overview.

Thank you so much, Ferenc, for your feedback. I'm feeling happy to see the code is good.

File: react/week4/github-user-searcher/src/components/UserSearcherProvider.jsx (hunk 1 above)
Comment (Saidemm) [reply react/week4/github-user-searcher/src/components/UserSearcherProvider.jsx:38]: Thanks. Perfect.

File: react/week4/github-user-searcher/src/components/UsersSearcher.jsx (hunk 2 above)
Comment (Saidemm) [reply react/week4/github-user-searcher/src/components/UsersSearcher.jsx]: You are completely right. I made it temporary but I forgot to change it. I'll make a descriptive name. Thanks

File: react/week4/github-user-searcher/src/components/UsersSearcher.jsx (hunk 3 above)
Comment (Saidemm) [reply react/week4/github-user-searcher/src/components/UsersSearcher.jsx]: Thanks a lot for testing it. Now I feel relieved.

**Pull request merged**